march_joint_inertia_controller/src/inertia_estimator.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright 2020 Project March.
+#include <algorithm>
+#include <cmath>
+#include <hardware_interface/joint_command_interface.h>
+#include <hardware_interface/joint_state_interface.h>
+#include "march_joint_inertia_controller/inertia_estimator.h"
+#include "ros/ros.h"
+#include "std_msgs/Float64.h"
+
+std::list<double> absolute(const std::list<double>& a)
+{
+  std::list<double> b;
+  for (auto it = a.begin(); it != a.end(); it++)
+  {
+    b.push_back(abs(*it));
+  }
+  return b;
+}
+
+double mean(const std::list<double>& a)
+{
+  double sum = 0.0;
+  for (auto it = a.begin(); it != a.end(); it++)
+  {
+    sum += *it;
+  }
+  if (a.size() == 0)
+  {
+    return 0.0;
+  }
+  return sum / a.size();
+}
+
+InertiaEstimator::InertiaEstimator(double lambda, size_t acc_size)
+{
+  lambda_ = lambda;
+  acc_size_ = acc_size;
+  joint_inertia_ = 0.0;
+
+  // Size the buffers
+  z1a.resize(sos_.size(), 0.0);
+  z2a.resize(sos_.size(), 0.0);
+
+  z1t.resize(sos_.size(), 0.0);
+  z2t.resize(sos_.size(), 0.0);
+
+  velocity_array_.resize(vel_size_, 0.0);
+
+  acceleration_array_.resize(acc_size_, 0.0);
+
+  filtered_acceleration_array_.resize(acc_size_, 0.0);
+
+  joint_torque_.resize(torque_size_, 0.0);
+
+  filtered_joint_torque_.resize(fil_tor_size_, 0.0);
+}
+
+double InertiaEstimator::getAcceleration()
+{
+  auto a = *acceleration_array_.begin();
+  return a;
+}
+
+void InertiaEstimator::setLambda(double lambda)
+{
+  lambda_ = lambda;
+}
+void InertiaEstimator::setAccSize(size_t acc_size)
+{
+  acc_size_ = acc_size;
+
+  velocity_array_.resize(acc_size_, 0.0);
+  acceleration_array_.resize(acc_size_, 0.0);
+  filtered_acceleration_array_.resize(acc_size_, 0.0);
```

Comment:
I don't know if you are aware of this but this does not reset the entire list to `0.0`. Only if new values need to be added, when the list is too small, will new values be created. Might not be a problem for you :shrug:

Reply:
I'm guessing that, as this is only used during initialization from lists without size, this won't be a problem.